app/services/email_checker.rb

```diff
@@ -83,4 +86,8 @@ def mx_records?
       Rails.configuration.mx_checker.records?(domain)
     end
   end
+
+  def sendgrid_api
+    @sendgrid_api ||= SendgridApi.new
```

Comment:
I would prefer that the API client was created by an initialiser (or at least only once). That way the persistent HTTP connection can be reused for multiple requests. This code opens a new connection for each rails request.